lib/middleware/metrics.rb

```diff
@@ -28,6 +28,18 @@ def is_private_ip?(env)
       !!(ip && ip.to_s =~ PRIVATE_IP)
     end
 
+    def is_trusted_ip?(env)
+      if
+        GlobalSetting.prometheus_trusted_ip_whitelist_regex.length == 0
+        nil
+      else
+        trusted_ip_regex = Regexp.new GlobalSetting.prometheus_trusted_ip_whitelist_regex rescue nil
```

Comment:
there is a lot of rescue nil here ... instead I guess if we need to rescue all ... put it all in a block ... and then if anything is raised just return false ... cause it is not trusted.